subscriptions/tasks.py

```diff
@@ -395,28 +344,70 @@ def create_initial_payload(self, to_addr, subscription):
 
         return payload
 
-    def get_message_to_send(self, subscription):
+
+send_current_message_inner = SendCurrentMessage()
+
+
+@app.task
+def pre_send_process(subscription_id, resend_id=None):
+    context = {"subscription_id": subscription_id}
+
+    if resend_id:
+        context["resend_id"] = resend_id
+
+    logger.info("Loading Subscription")
+    subscription = Subscription.objects.get(id=context["subscription_id"])
```

Comment:
Can we do a `select_related` here so that we don't need an additional DB request when we're getting the messageset